### PR TITLE
Update the run_demo script to add background mode and usage info

### PR DESCRIPTION
**Pull request by swcurran**

### CODE REVIEW

File: demo/run_demo [hunk 1]
```diff
@@ -46,6 +50,38 @@ do
 			echo "For example, to create the directory and then set the permissions use: 'mkdir ../logs; chmod uga+rws ../logs'"
 			exit 1
 		fi
+		continue
```
Comment (andrewwhitehead) [inline demo/run_demo]: Adding continue here means that the argument gets stripped off, which I don't think is intended.

File: demo/run_demo [hunk 2]
```diff
@@ -46,6 +50,38 @@ do
...
+		;;
+	--bg)
+		if [[ "${AGENT}" == "alice" ]] || [[ "${AGENT}" == "faber" ]] || [[ "${AGENT}" == "acme" ]]; then
+			DOCKER_OPTS="-d"
+			echo -e "\nRunning in ${AGENT} in the background. Note that you cannot use the comand line console in this mode."
```
Comment (andrewwhitehead) [inline demo/run_demo]: *command :)

File: demo/run_demo (hunk 1 above)
Comment (swcurran) [reply demo/run_demo]: Thanks - wasn't sure about that.

Comment (swcurran) [reply demo/run_demo]: Fixed

File: demo/run_demo (hunk 2 above)
Comment (swcurran) [reply demo/run_demo]: Fixed

File: demo/run_demo (hunk 1 above)
Comment (swcurran) [reply demo/run_demo]: Confirmed that --timing works.

**Pull request merged**